src/extra/gfx/steps.cpp

```diff
@@ -343,42 +355,40 @@ struct EffectPassShard {
   SHVar activate(SHContext *context, const SHVar &input) {
     RenderFullscreenStep &step = std::get<RenderFullscreenStep>(*_step->get());
 
-    checkType(input.valueType, SHType::Table, "Input table");
-    const SHTable &inputTable = input.payload.tableValue;
-
-    shared::applyAll(context, step, inputTable);
+    // Basically applyAll but not with table input. Can extract out into a function
+    shared::applyAll(context, step, _outputs, _outputScale, _features);
+    // End of applyAll
 
-    SHVar inputsVar;
-    if (getFromTable(context, inputTable, "Inputs", inputsVar))
-      applyInputs(context, step, inputsVar);
-    else {
+    if (!_inputs.isNone()) {
+      applyInputs(context, step, _inputs.get());
+    } else {
       step.inputs = RenderStepInputs{"color"};
     }
 
-    SHVar paramsVar;
-    if (getFromTable(context, inputTable, "Params", paramsVar))
-      applyParams(context, step, paramsVar);
+    if (!_params.isNone()) {
+      applyParams(context, step, _params.get());
+    }
 
     // NOTE: First check these variables to see if we need to invalidate the feature Id (to break caching)
-    SHVar composeWithVar;
-    if (getFromTable(context, inputTable, "ComposeWith", composeWithVar)) {
-      // Always create a new object to force shader recompile
-      wrapperFeature = std::make_shared<Feature>();
-      applyComposeWith(context, composeWithVar);
-    }
+    // TODO: Check if there should be a ComposeWith parameter in the first place
+    // SHVar composeWithVar;
+    // if (getFromTable(context, inputTable, "ComposeWith", composeWithVar)) {
+    //   // Always create a new object to force shader recompile
+    //   wrapperFeature = std::make_shared<Feature>();
+    //   applyComposeWith(context, composeWithVar);
+    // }
```

Comment:
For `EffectPassShard`, it appears to have expected a `ComposeWith` parameter from the input table in the past and has an applyComposeWith function, but this parameter is not mentioned in the original possible expected table format and .edn test files involving `EffectPassShard` have never called it with a `ComposeWith` key in its input table before. 

Thus need to clarify if this shard actually needs a ComposeWith parameter, or if the ComposeWith code was mistakenly leftover from modifying a copy of `DrawablePassShard`

Reply:
Yeah, it's not in any of the shards samples but an example can be found in the [fragcolor-xyz/rare-examples/gfx/gfx-master-pbr.edn](https://github.com/fragcolor-xyz/rare-examples/blob/6543ebf517a53507ab878f2ee0f91d282df5315d/gfx/gfx-master-pbr.edn#L35)
```edn
...
{:ComposeWith {:pbr-wgsl .file-pbr-wgsl.json}
   :EntryPoint (->
                (Shader.Literal :Type ShaderLiteralType.Header :Source .pbr-wgsl)
                (Shader.ReadInput "texCoord0") >= .uv
                (Shader.Literal :OutputDimension 2 :Source ["computeLUT(" .uv ", 1024)"])
                (ToFloat2) (Shader.WriteOutput "color") ; End shader
                .uv (Shader.WriteOutput "color"))}
  (GFX.EffectPass) >> .lut-steps-1
...
```